README-OTP2-TERMINOLOGY.md

```diff
@@ -0,0 +1,117 @@
+- arrival time (seconds)
+- number of transfers (scalar)
+- travel duration (seconds)
+- cost (scalar) (a function of anything else than mention above)
+
+We will also experiment with extracting other criteria from the _cost_ like _walkingDistance_,  _operator_. The goal is to make this configurable so each deployment may tune this to their needs. Due to performance reasons it might not be 100% dynamic.
+
+Because _Raptor_ is much faster than the _multi-criteria_ Raptor we will provide an option (request parameter) to run both _RR_  and _McRR_. We might even use _RR_ as a heuristics optimization for McRR. In a bench mark test, RR takes on average 80ms while McRR with the same configuration takes 400ms. If we add _walking distance_ as an extra criteria the average time increase to 1000ms. (The numbers provided are for relative comparison - it is to early to compare with the existing OTP performance).
+
+## Pareto optimal/efficiency
+All paths that is considered *pareto optimal* for a set of criteria is returned by the StdRR and McRR, but in most cases were we talk about pareto optimal paths, we talk about the McRR version - adding _cost_ or some other criteria. The set of pareto optimal paths can be a large set (> 500 paths).
+
+### Pareto-set explained
+See [Wikipedia](https://en.wikipedia.org/wiki/Pareto_efficiency) A pareto-set of paths/itineraries is a set where all elements are better (less than) for at least one criteria than all other elements in the set. Given a set `{ [9,2], [5,6],  [3, 8] }` then `[7, 4]` would make it into the set. This is because  7 < 9 (comparing the 1sr criteria, element 1), and 4 < 6 and 8 (comparing the 2nd criteria, element 2 and 3). `[6,7]` would not make it into the set because `[5,6]` is better for both criteria. 
+
+### Comparison, Dominance and Mutual dominance
+When two criteria/values are compared we say that a value _dominates_ another value if it is better. A vector dominates another, if one of its criteria dominates the corresponding criteria of the other vector. Two vectors are mutual dominant if they dominates each other, both entities are pareto-optimal. Lets say _less_ is better, then [1,5] and [4,2] mutually dominates each other. This is because 1 < 4 (1st value) and 5 > 2 (2nd value). We can also have mutual dominance between to criteria, if we compare using `!=` or _Relaxed comparison_ the comparison is mutual dominant. 
+
+### Relaxed comparison
+A comparison may be *relaxed*, that is, a slack or delta is added on one side of the comparison to make it easier for a value to dominate. Let say _cost_ is one of the criteria in the vector, but we want to accept not only pareto optimal paths, but all paths that have a cost that is only 10% worse. We can do so by using a comparison like this `c' <= 1.10 * c"`. If we use this 10% slack the value 10 dominates 11: `10 <= 11 * 1.1`. But, also 11 dominates 10, because: `11 <= 10 * 1.1`. 
+
+# The _API_ package
```

Comment:
I had started moving these package descriptions, and some class descriptions, into Javadoc and `package-info.java` files. Once these `README` files are checked in, I will propose those changes as a separate PR.